src/libsyntax_ext/proc_macro_registrar.rs

```diff
@@ -84,69 +94,19 @@ pub fn modify(sess: &ParseSess,
     return krate
 }
 
-impl<'a> CollectCustomDerives<'a> {
+impl<'a> CollectProcMacros<'a> {
     fn check_not_pub_in_root(&self, vis: &ast::Visibility, sp: Span) {
         if self.is_proc_macro_crate &&
            self.in_root &&
            *vis == ast::Visibility::Public {
             self.handler.span_err(sp,
                                   "`proc-macro` crate types cannot \
                                    export any items other than functions \
-                                   tagged with `#[proc_macro_derive]` \
-                                   currently");
+                                   tagged with `#[proc_macro_derive]` currently");
```

Comment:
No, this might be printed before we've ever discovered any items with `#[proc_macro_derive]` or `#[proc_macro_attribute]`. You didn't want to mention both before, either.